### DISCUSSION TITLE
ImageMagick stuck in "montage loop" with a large set of images

### DISCUSSION THREAD

**Discussion by muuyo** [ImageMagick/ImageMagick#7443] [Help] [answered]
### ImageMagick version

7.1.1-Q16-HDRI

### Operating system, version and so on

Windows 11 23H2

### Description

I'm attempting to take ~65,000 images into a tiled flipbook spritesheet style thing for use in Unreal Engine, however when I try to run the command to montage the folder, it simply loops like this.
![07_01 01-40-31](https://github.com/ImageMagick/ImageMagick/assets/34991771/b2ce3d1e-9663-429a-9402-4dc41186804e)
Named like so.
![image](https://github.com/ImageMagick/ImageMagick/assets/34991771/35b875d7-2c69-439a-9632-5dd15258831b)
If anyone has any recommendations as to alternate ways to do this, *especially* ones that could divvy it up somehow (since this is *hammering* my SSD, I'm all ears as well in case the issue is convoluded. I'm just not a talented enough scripter to figure that much out myself.
It's been doing this for about fifteen minutes now, and I realize it's a large set, but that feels like what this is designed for.

**Comment by snibgo** [comment, marked as answer]
How large are the inputs, in pixels? Are they the same size, or different sizes? Did you kill the process? How long did you leave it running?

A simple experiment suggests there may a problem in the `-monitor` output, in the `Composite/Image` stage, but the process does eventually finish, for me, unless I run out of memory.
```
magick montage toes.png -duplicate 5999 -geometry +0+0 -tile 8x9 -monitor x.jpg
```
... where toes.png is 267x233 pixels. This takes 68 seconds for me. When I `-duplicate 59999`, I get an error:

```
montage: unable to allocate string `Not enough space' @ fatal/string.c/AcquireString/112.
```

You can divide the job like this:
```
magick montage bad_apple_[0001-0100].png -geometry +0+0 -tile 8x9 -monitor x.jpg
magick montage bad_apple_[0101-0200].png -geometry +0+0 -tile 8x9 -monitor x.jpg
{... and so on}
```

**Reply by muuyo** [reply]
yeah, turns out that even though the display is broken it ended up outputting afterwards. I did end up deleting half the frames (so 4000) and reducting the resolution to 720x540, so now I have a nice clean 45360x33480 image.
